packages/algorand-js/src/types.ts

```diff
@@ -13,58 +13,145 @@ export type AppArgs = Array<string | number>;
 export type StackElem = bigint | Uint8Array;
 export type TEALStack = IStack<bigint | Uint8Array>;
 
-// https://developer.algorand.org/docs/reference/teal/opcodes/#txn
-export enum TxnField {
-  Sender, // 32 byte address
-  Fee, // micro-Algos
-  FirstValid, // round number
-  FirstValidTime,
-  LastValid, // round number
-  Note,
-  Lease,
-  Receiver, // 32 byte address
-  Amount, // micro-Algos
-  CloseRemainderTo, // 32 byte address
-  VotePK, // 32 byte address
-  SelectionPK, // 32 byte address
-  VoteFirst,
-  VoteLast,
-  VoteKeyDilution,
-  Type,
-  TypeEnum,
-  XferAsset, // Asset ID
-  AssetAmount, // value in Asset's units
-  AssetSender, // 32 byte address. Causes clawback of all value of asset from AssetSender if Sender is the Clawback address of the asset.
-  AssetReceiver, // 32 byte address
-  AssetCloseTo, // 32 byte address
-  GroupIndex, // Position of this transaction within an atomic transaction group. A stand-alone transaction is implicitly element 0 in a group of 1
-  TxID, // The computed ID for this transaction. 32 bytes.
-  ApplicationID, // ApplicationID from ApplicationCall transaction.
-  OnCompletion, // ApplicationCall transaction on completion action.
-  ApplicationArgs, // Arguments passed to the application in the ApplicationCall transaction.
-  NumAppArgs, // Number of ApplicationArgs.
-  Accounts, // Accounts listed in the ApplicationCall transaction.
-  NumAccounts, // Number of Accounts
-  ApprovalProgram, // Approval program
-  ClearStateProgram, // Clear state program
-  RekeyTo, // 32 byte Sender's new AuthAddr
-  ConfigAsset, // Asset ID in asset config transaction
-  ConfigAssetTotal, // Total number of units of this asset created
-  ConfigAssetDecimals, // Number of digits to display after the decimal place when displaying the asset
-  ConfigAssetDefaultFrozen, // Whether the asset's slots are frozen by default or not, 0 or 1
-  ConfigAssetUnitName, // Unit name of the asset
-  ConfigAssetName, // The asset name
-  ConfigAssetURL, // URL
-  ConfigAssetMetadataHash, // 32 byte commitment to some unspecified asset metadata
-  ConfigAssetManager, // 32 byte address
-  ConfigAssetReserve, // 32 byte address
-  ConfigAssetFreeze, // 32 byte address
-  ConfigAssetClawback, // 32 byte address
-  FreezeAsset, // Asset ID being frozen or un-frozen
-  FreezeAssetAccount, // 32 byte address of the account whose asset slot is being frozen or un-frozen.
-  FreezeAssetFrozen // The new frozen value, 0 or 1
+export interface TxnEncodedObj {
```

Comment:
can we move this to types-algosdk ? or is it more relevant to put it here. what do you think ?

Reply:
yes, i was thinking the same. Will need to confirm with Robert. in types/algosdk i think 'unknown' is being used.

Reply:
Yes, very good call. Let's put it to the `types-algosd`.